src/openms/source/FORMAT/OMSFileStore.cpp

```diff
@@ -345,8 +307,16 @@ namespace OpenMS::Internal
     for (const String& info_key : info_keys)
     {
       query.bindValue(":name", info_key.toQString());
-      Key value_id = storeDataValue_(info.getMetaValue(info_key));
-      query.bindValue(":data_value_id", value_id);
+      const DataValue& value = info.getMetaValue(info_key);
+      if (value.isEmpty()) // use NULL as the type for empty values
+      {
+        query.bindValue(":data_type_id", QVariant(QVariant::Int));
```

Comment:
Yes, that's how NULLs are done using Qt.
It may be worth adding a comment to the "..._MetaInfo" table definition code (around line 278 in "OMSFileStore.cpp") to say that empty values are represented using NULL `data_type_id`.